src/Action.php

```diff
@@ -60,6 +60,28 @@ public function __construct(BaseAction &$action)
     }
 
     /**
+     * Returns the path for an action source file.
+     *
+     * @param string $filename
+     * @return string
+     * @throws \Exception
+     */
+    private function getSourcePath(string $filename): string
+    {
+        $path = implode(DIRECTORY_SEPARATOR, [
```

Comment:
I would add a `log()` with "[TAO]" after this, to capture the path being resolved.